### ISSUE TITLE
False positive for `unicorn/no-array-method-this-argument` and `unicorn/no-array-callback-reference` when using leaflet's `L.map();`

### ISSUE THREAD

**Issue by puka-tchou** [sindresorhus/eslint-plugin-unicorn#1699]
The rules `unicorn/no-array-method-this-argument` and `unicorn/no-array-callback-reference` return a false positive with leaflet's `L.map()` function.

`unicorn/no-array-method-this-argument`
`unicorn/no-array-callback-reference`

```js
import L from "leaflet";

const map = (element, options) => {
  const map = L.map(element, options); // this should be OK
}
// the false positive are:
// "Do not pass function `element` directly to `.map(…)`."
// "Do not use the `this` argument in `Array#map()`."
```
Additional info: https://leafletjs.com/reference.html#map-example

**Comment by fregante** [comment]
Tracked in:
- https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1394
- https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1193

respectively
